tests/dummy/app/snippets/cart-contents-lookup.ts

```diff
@@ -0,0 +1,18 @@
+// BEGIN-SNIPPET cart-contents-lookup.ts
+import Component from '@glimmer/component';
+import { getOwner } from '@ember/application';
+import { action } from '@ember/object';
+
+import ShoppingCartService from 'my-app/services/shopping-cart';
+
+export default class CartContentsComponent extends Component {
+  get cart() {
+    return getOwner(this).lookup('service:shopping-cart') as ShoppingCartService;
```

Comment:
You mentioned it in the writeup, but let's be *extra* explicit that this cast is wildly unsafe and the user has to guarantee it. If they *typo* the lookup, they'll get back `null` or something else. (It occurs to me that we could improve this by defining an `OwnerRegistry` and registering things there; mostly I wish that the `lookup` API were `lookup(bucket, name)` instead of `lookup('<bucket>:<name>')`, because if it were we could trivially reuse the existing registries.

Reply:
Definitely would be good to be extra explicit here. I was thinking the exact same thing about a registry for this.

Reply:
I opened [an issue on the RFCs repo](https://github.com/emberjs/rfcs/issues/583) for that idea, and will hopefully 🤞 write an actual RFC for it if there's interest.

Reply:
[RFC #585](https://github.com/emberjs/rfcs/pull/585) for the above concern!